src/lib/sfge/SfgePathlessEngine.ts

```diff
@@ -39,14 +39,9 @@ export class SfgePathlessEngine extends AbstractSfgeEngine {
 				return true;
 			}
 		}
-		const haltString = messages.getMessage('errors.failedWithoutProjectDir', []);
-		const warnString = messages.getMessage('warnings.skippedWithoutProjectDir', [
-			this.getName(),
-			'missingOptionsBehavior',
-			MissingOptionsBehavior.WARN,
-			this.config.getConfigFilePath()
-		]);
-		return this.handleMissingOptionsBehavior(this.missingOptionsBehavior, haltString, warnString);
+		// If we're here, it's because we're missing the necessary info to run this engine.
+		// We should throw an error indicating this.
+		throw SfdxError.create('@salesforce/sfdx-scanner', 'SfgeEngine', 'errors.failedWithoutProjectDir', []);
```

Comment:
Provides expected behavior of throwing exception if `--projectdir` isn't present.